src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/ValueResolver/DateTimeValueResolver.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Controller\ArgumentResolver\ValueResolver;
+
+use Symfony\Component\ArgumentResolver\ArgumentMetadata\ArgumentMetadata;
+use Symfony\Component\ArgumentResolver\Exception\InvalidArgumentException;
+use Symfony\Component\ArgumentResolver\ValueResolver\Traits\DateTimeValueResolverTrait;
+use Symfony\Component\HttpFoundation\Request;
+
+/**
+ * Convert DateTime instances from request attribute variable.
+ *
+ * @author Benjamin Eberlei <kontakt@beberlei.de>
+ * @author Tim Goudriaan <tim@codedmonkey.com>
+ */
+final class DateTimeValueResolver implements ControllerValueResolverInterface
+{
+    use DateTimeValueResolverTrait;
+
+    public function resolve(ArgumentMetadata $argument, ?Request $request = null): iterable
+    {
+        if (!$request) {
```

Comment:
This new interface forcing all implementations to handle the case of a `null` request looks like a strictly inferior solution compared to the existing feature (and I don't see the benefit for usage in the console component, as ControllerValueResolverInterface won't be reused in the console, so we could simply implement a similar system in symfony/console without changing the system of HttpKernel)

Reply:
While it's not ideal, I don't think it's a big deal. This could even be addressed by replacing the instanceof check by phpdoc, which I think I'm going to do.